rust/yarp-sys/build.rs

```diff
@@ -0,0 +1,168 @@
+use std::{
+    io::{self, Write},
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+fn main() -> Result<(), io::Error> {
+    rake_compile()?;
+
+    let ruby_build_path = ruby_build_path();
+    let ruby_include_path = ruby_include_path();
+
+    // Tell cargo/rustc that we want to link against `librubyparser.a`.
+    println!("cargo:rustc-link-lib=static=rubyparser");
+
+    // Add `[root]/build/` to the search paths, so it can find `librubyparser.a`.
+    println!(
+        "cargo:rustc-link-search=native={}",
+        ruby_build_path.display()
+    );
+
+    // This is where the magic happens.
+    let bindings = generate_bindings(&ruby_include_path);
+
+    // Write the bindings to file.
+    write_bindngs(bindings);
+
+    Ok(())
+}
+
+/// Convenience step for running `rake compile` before doing our stuff.
+///
+fn rake_compile() -> Result<(), io::Error> {
+    let output = Command::new("rake").arg("compile").output()?;
+
+    io::stdout().write_all(&output.stdout)?;
+    io::stderr().write_all(&output.stderr)?;
+
+    println!("`rake compile` exit status: {}", output.status);
+
+    Ok(())
+}
+
+/// Gets the path to project files (`librubyparser*`) at `[root]/build/`.
+///
+fn ruby_build_path() -> PathBuf {
+    cargo_manifest_path()
+        .join("../../build/")
+        .canonicalize()
+        .unwrap()
+}
+
+/// Gets the path to the header files that `bindgen` needs for doing code generation.
+///
+fn ruby_include_path() -> PathBuf {
+    cargo_manifest_path()
+        .join("../../include/")
+        .canonicalize()
+        .unwrap()
+}
+
+fn cargo_manifest_path() -> &'static Path {
+    Path::new(env!("CARGO_MANIFEST_DIR"))
```

Comment:
This env variable is set when the build script is run, not when it is compiled. It must be resolved with `std::env::var_os` at runtime.

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts

Reply:
Ah, right! Nice catch!